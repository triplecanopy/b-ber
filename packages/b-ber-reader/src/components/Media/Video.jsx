import React from 'react'
import Media from './Media'
import VideoElement from './VideoElement'

const Video = props => (
  <Media mediaType="video" MediaComponent={VideoElement} {...props} />
)

export default Video
