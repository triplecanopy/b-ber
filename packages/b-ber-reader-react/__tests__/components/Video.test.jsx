/* eslint-disable react/jsx-props-no-spreading */

import React from 'react'
import { render } from '@testing-library/react'
import Video from '../../src/components/Media/Video'

jest.mock(
  '../../src/lib/with-node-position',
  () => WrappedComponent => props => <WrappedComponent {...props} />
)

describe('Video', () => {
  // Clean up logging from HOC during tests
  beforeEach(() => {
    console.warn = jest.fn()
    console.error = jest.fn()
  })

  afterEach(() => jest.clearAllMocks())

  test('renders the component', () => {
    let props
    let tree

    const ref = React.createRef()

    props = { id: 'foo', 'data-autoplay': true, controls: true }
    tree = render(<Video elemRef={ref} {...props} />)

    expect(tree.container).toMatchSnapshot()

    props = { id: 'foo', 'data-autoplay': false, controls: false }
    tree = render(<Video elemRef={ref} {...props} />)

    expect(tree.container).toMatchSnapshot()
  })
})
