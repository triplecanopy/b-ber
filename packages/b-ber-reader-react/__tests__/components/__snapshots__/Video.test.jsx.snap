// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders the component 1`] = `
<div>
  <video
    controls=""
    data-autoplay="true"
    id="foo"
  />
</div>
`;

exports[`Video renders the component 2`] = `
<div>
  <video
    data-autoplay="false"
    id="foo"
  />
</div>
`;
